docs: remove some dependencies on old setup, redirect setup to new setup-local

jenniferfell opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
"Setup for local development" (setup.md) contains legacy information. Specifically, it describes how to use the old **deprecated** QuickStart repo and example application (pre CLI) to start a new project. We can't yet remove this document because the QuickStart app is referenced by other guides: 

* Upgrading from AngularJS
* Updating for Performance
* Typescript configuration
* Testing
* Change log (slated to be removed anyway)

When all references have been updated, then this guide can be removed. 

For now, it is hidden from the navigation, but it does show up in search (1st result for "setup"). 

Issue Number: N/A


## What is the new behavior?
Adding a legacy note to the title and intro part of the document will help users know:
* when to use this information
* what to use instead

Begin to remove information that is more accurate elsewhere. 
* [node and npm appendix](https://angular.io/guide/setup#appendix-nodejs-and-npm) is now covered by prereqs in Getting Started
* [why develop locally appendix](https://angular.io/guide/setup#appendix-why-develop-locally) is unnecessary. The intro sentences are sufficient.
* [developing with IE](https://angular.io/guide/setup#appendix-develop-locally-with-ie) might also be unnecessary. If it is a permanent issue, it should be moved to [browser support.](https://angular.io/guide/browser-support) 

Remove the emphasis about stackblitz not being where you would develop a "real" app. This isn't the positioning that we want today. 

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

jenniferfell committed:
docs: position old setup guide as legacy until it can be removed

mary-poppins commented:
You can preview 9ed920c at https://pr26758-9ed920c.ngbuilds.io/.

jenniferfell commented:
Note to self: Some updates to the repo's readme might enable us to remove setup.md sooner, and redirect links to that repo's readme where we still have dependencies from angular.io

mary-poppins commented:
You can preview 4a4d7a7 at https://pr26758-4a4d7a7.ngbuilds.io/.

mary-poppins commented:
You can preview 9ed920c at https://pr26758-9ed920c.ngbuilds.io/.
You can preview 7b198fc at https://pr26758-7b198fc.ngbuilds.io/.
You can preview 4a4d7a7 at https://pr26758-4a4d7a7.ngbuilds.io/.
You can preview 80610b9 at https://pr26758-80610b9.ngbuilds.io/.
You can preview 9604cf5 at https://pr26758-9604cf5.ngbuilds.io/.
You can preview d8907af at https://pr26758-d8907af.ngbuilds.io/.
You can preview ee4a2ea at https://pr26758-ee4a2ea.ngbuilds.io/.
You can preview b29b907 at https://pr26758-b29b907.ngbuilds.io/.

mary-poppins commented:
You can preview bc0fc77 at https://pr26758-bc0fc77.ngbuilds.io/.

mary-poppins commented:
You can preview d81ff72 at https://pr26758-d81ff72.ngbuilds.io/.

jenniferfell commented:
Recent updates: 
* Renamed file from `setup.md` to `upgrade-setup.md` so that we can potentially use `setup.md` for its original purpose as the main file for local setup instructions.
* Rewrote introductory audience/purpose statement in `upgrade-setup.md` to position that information as useful only in the context of the 2 upgrading guides. 
* Updated links from the 2 upgrading guides to `upgrade-setup.md`.
* Updated links from `change-log.md` to `upgrade-setup.md`. (Won't build with bad links, even if there is another PR to remove that doc.)
* Removed dependencies on this guide from two other guides: 
- Replaced TypeScript Config example of `tsconfig.json` with one from a current CLI app instead of the old one here from the deprecated QuickStart repo. 
- Replaced Testing guide's links to this setup doc with text explaining the need to import `zone-testing`. The CLI does it now, so the longer explanation isn't needed.

gkalpak commented on aio/content/navigation.json:
```diff
@@ -576,12 +582,6 @@
       "tooltip": "Build, testing, and deployment environment, tool, and configuration information.",
       "children": [
-        {
-          "url": "guide/setup",
```
What about the broken links to `/guide/setup`?

petebacondarwin replied:
Good spot. Do we need a redirect?

jenniferfell replied:
I fixed all links from within AIO. 

For external links, I think the cleanest thing long-term to cause most external links to go to the intended kind of content is this: 

1. Remove `quickstart.md` and redirect that to `start/index.md` (the new getting started). See PR https://github.com/angular/angular/pull/29651

2. Move the very old setup content from `setup.md` into a new file called `upgrade-setup.md` because that is the only context in which that information should still be used (that's this PR)

3. Put current local setup content into `setup.md`. PR https://github.com/angular/angular/pull/29651 puts that content into `setup-local.md` which is fine, but if we reuse the old `setup.md` filename, then external links will do the reasonable thing and go to the most current local setup information. No PR for this change yet, but I could work it into PR 29651 or this one. 

A final thought: Most of the info is in the readme for the very old QuickStart repo. We could just remove this upgrade-setup.md entirely, link to the repo's readme for those who still need that old setup information, and then either use setup.md for current info or redirect to setup-local.md.

jenniferfell replied:
I added the question on the QS PR as well re refactoring setup info now, rather than creating a new setup-local.md, when we can reuse setup.md for the current info and move the old info into upgrade-setup.md.

mary-poppins commented:
You can preview cf1ef70 at https://pr26758-cf1ef70.ngbuilds.io/.

jenniferfell commented:
To do -- either: 
* Move this setup repositioning and refactoring into https://github.com/angular/angular/pull/29651
OR
* Fix link in the first alert box, so that it goes to the correct location  after that PR lands.

mary-poppins commented:
You can preview 64897c5 at https://pr26758-64897c5.ngbuilds.io/.

mary-poppins commented:
You can preview 1d22702 at https://pr26758-1d22702.ngbuilds.io/.

mary-poppins commented:
You can preview ada92d9 at https://pr26758-ada92d9.ngbuilds.io/.

jenniferfell committed:
docs: add redirect from old setup.md to new setup-local.md

mary-poppins commented:
You can preview 99a4211 at https://pr26758-99a4211.ngbuilds.io/.

jenniferfell commented:
Added redirect from old `setup.md` to new `setup-local.md` (PR https://github.com/angular/angular/pull/29651) with the assumption that we'll have this...

Old:
* `quickstart.md` contained local setup information and a quick tutorial
* `setup.md` contained legacy local setup information that uses the old QuickStart repo and is still referenced by the upgrade docs

New:
* `quickstart.md` redirects to `start/index.md` which is the new getting started. Places within the doc set that linked to `quickstart.md` for setup information now link to `setup-local.md`.
* `setup.md` redirects to `setup-local.md` which has current local setup info. Places within the doc set that linked to `setup.md` for legacy setup information now link to `upgrade-setup.md`.

So the old filenames are completely removed, rather than reused for new purposes. This is good for `quickstart` which was heavily overloaded as a phrase and title. 


**Note:** This PR won't build without errors and will need to be rebased after https://github.com/angular/angular/pull/29651 merges.

mary-poppins commented:
You can preview f879496 at https://pr26758-f879496.ngbuilds.io/.

mary-poppins commented:
You can preview a8b0d68 at https://pr26758-a8b0d68.ngbuilds.io/.

jenniferfell commented:
@IgorMinar and @gkalpak : Please review. This PR does two things (separated into two commits): 

* Remove as many dependencies as possible on the old legacy setup.md, which references the old QuickStart repo. 

* Rename setup.md to upgrade-setup.md, so we can redirect setup.md to the new setup-local.md, which is probably where readers want to end up. (I've removed all internal xrefs, but external sites might still link to the old setup.md.) Also, this makes the purpose more clear in the URL, not just in the page title. 

Future work: Rewrite upgrading docs so they no longer depend on this old setup information, or redirect those upgrade docs to just use the readme.md in the quickstart repo.

gkalpak commented on aio/content/guide/testing.md:
```diff
@@ -1253,6 +1254,14 @@ The `fakeAsync()` function enables a linear coding style by running the test bod
+
+<div class="alert is-helpful">
+
+Limitation: The `fakeAsync()` function won't work if the test body makes an XHR call. XHR calls within a test are rare,but if you need to call XHR, see [`async()`](#async), below.
```
Missing space after `rare,`.

gkalpak replied:
I know there are other references of `XHR` here, but:
1. It might be confusing for beginners (not knowing that `XHR` stands for `XMLHttpRequest` or even what `XMLHttpRequest` is :grin:)
2. Does the same limitation apply to using the `fetch` API (instead of XHR)? If so, it should probably be mentioned too.

gkalpak commented on aio/content/guide/upgrade-performance.md:
```diff
@@ -216,7 +216,7 @@ the recipe.
 In order to start using any `upgrade/static` APIs, you still need to load the Angular framework as
 you would in a normal Angular app. You can see how this can be done with SystemJS by following the
-instructions in the [Setup](guide/setup) guide, selectively copying code from the
+instructions in the [Upgrade Setup](guide/upgrade-setup "Setup for Upgrading to AngularJS") guide, selectively copying code from the
```
_Upgrading to AngularJS_ --> _Upgrade from AngularJS_ (?)

(In other `guide/upgrade-setup` references too.)

gkalpak commented on aio/content/navigation.json:
```diff
@@ -506,6 +506,12 @@
+            {
+              "url": "guide/upgrade-setup",
+              "title": "Setup for Upgrading to AngularJS",
+              "tooltip": "Use code from the Angular QuickStart seed as part of upgrading to AngularJS.",
```
_"Upgrading **to** AngularJS"_ (here and above) :smiley:
I know it is a hidden item, but the placement is a little weird. Consider putting it above `guide/upgrade`.

gkalpak reviewed (approved):
Left a couple minor comments, but lgtm otherwise 👌 
Also: Yay! More redirection rules 😛

jbogarthyde commented:
Content transferred to new PR https://github.com/angular/angular/pull/32193
